extensions/helper/Assets.php

```
@@ -43,71 +28,23 @@ public function style($manifest) {
 
 	/**
 	 * Takes a manifest name and:
-	 * 1. compiles coffee files to js
-	 * 2. mangles all files in the manifest (if $config['mangle'] is true)
-	 * 3. returns one or more script tags (based on mangle setting)
+	 * 1. Compiles coffee files to js.
+	 * 2. Merges all files in the manifest (if $config['mergeAssets'] is true).
+	 * 3. Returns one or more script tags.
 	 *
 	 * @param  string $manifest name of manifest
-	 * @return string           javascript script tag(s)
+	 * @return string javascript script tag(s)
 	 */
 	public function script($manifest) {
-		$files = $this->build('js', $manifest);
+		$manifest = Manifest::load('js', $manifest);
+		$files = $manifest->build();
 		$tags = array();
-		foreach($files as $file) {
+		foreach ($files as $file) {
 			$tags[] = $this->_context->helper('html')->script($file);
 		}
 		return implode("\n", $tags);
 	}
 
-	/**
-	 * Initializes a new Manifest and calls compile() on it.
-	 *
-	 * @param  string $type     either 'js' or 'css'
-	 * @param  string $manifest name of manifest
-	 * @return array            filenames of assets
-	 */
-	protected function build($type, $manifest) {
-		Manifest::resetProcessed();
-		$manifest = new Manifest($type, $manifest);
-		$manifest->timestamp = !$this->config['batch_only'];
-		$manifest->verbose = $this->debug;
-
-		if($this->config['batch_only']) {
-			$files = array($this->mangledFilename());
```

This right here scares the crap out of me. By my reasoning `$this->mangledFilename()` should throw an error in production. So how were these helpers ever working?!?!